scripts/test_emulator_file_transfer.sh

```
@@ -0,0 +1,70 @@
+curl -fsS -F "file=@$UPLOAD_FILE;filename=test.txt" "http://127.0.0.1:8080/upload?path=/inbox" >/dev/null
+curl -fsS "http://127.0.0.1:8080/download?path=/inbox/test.txt" >"$DOWNLOAD_FILE"
+cmp "$UPLOAD_FILE" "$DOWNLOAD_FILE"
+curl -fsSI "http://127.0.0.1:8080/download?path=/inbox/test.txt" | grep -qi '^Content-Length: 29'
+[ "$(curl -sS -o /dev/null -w '%{http_code}' -I "http://127.0.0.1:8080/download?path=/missing.txt")" = "404" ]
+
+curl -fsS -X POST -d 'path=/inbox/test.txt&name=renamed.txt' "http://127.0.0.1:8080/rename" >/dev/null
+curl -fsS -X MKCOL "http://127.0.0.1:8080/dav" >/dev/null
+printf 'alpha' | curl -fsS -X PUT --data-binary @- "http://127.0.0.1:8080/dav/a.txt" >/dev/null
+curl -fsS -X PROPFIND -H 'Depth: 1' "http://127.0.0.1:8080/dav" | grep -q 'a.txt'
+curl -fsSI -X OPTIONS "http://127.0.0.1:8080/dav" | grep -qi '^Allow: .*MOVE'
+! curl -fsSI -X OPTIONS "http://127.0.0.1:8080/dav" | grep -Eqi '^Allow: .*(LOCK|UNLOCK)'
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)